hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/HddsDatanodeService.java

```diff
@@ -125,6 +125,9 @@ public class HddsDatanodeService extends GenericCli implements ServicePlugin {
   private HddsDatanodeClientProtocolServer clientProtocolServer;
   private OzoneAdmins admins;
   private ReconfigurationHandler reconfigurationHandler;
+  // Default datanode initial and current version to be used when datanode.id file doesn't exist
+  // Note: Currently this is only used in tests.
+  private int defaultInitialVersion = -1, defaultCurrentVersion = -1;
```

Comment:
I don't think we should add these (and >= 0 logic + the get/set methods) to enable testing.  Change this class to access `DatanodeVersion.CURRENT_VERSION` via a static method, and then mock that method in tests where needed.

https://javadoc.io/doc/org.mockito/mockito-core/4.11.0/org/mockito/Mockito.html#48

Reply:
Thanks @adoroszlai .

I wanted to allow different datanodes to have different initial and current version for testing. Mockito can't seem to return different values for each instance.

I have switched to the approach you suggested for now.

Reply:
Another approach that I thought about even before adding these two new fields is to create custom datanode.id file for each DN before they boot up. But I considered that too hacky than the new fields approach so I didn't go that way.

Reply:
Thanks for the update.

FYI, cleanup of `MiniOzoneCluster`'s interface is in progress (HDDS-6739).  I plan to introduce a way to allow customizing each datanode (HDDS-10320).

Reply:
nice! Thanks @adoroszlai